### PR TITLE
Add runtime_overhead PR Time Benchmark

### PR DESCRIPTION
**Pull request by azahed98**
This adds a PR time benchmark that checks for runtime overhead on a very small graph. This will help track regressions in runtime overhead.

Example Results:
```
runtime_overhead_inductor,instruction_count,222645
runtime_overhead_inductor_inference_mode,instruction_count,234998
runtime_overhead_inductor_requires_grad,instruction_count,293556
runtime_overhead_inductor_requires_grad_backward,instruction_count,78181
runtime_overhead_inductor_dynamic,instruction_count,234870
runtime_overhead_inductor_inference_mode_dynamic,instruction_count,248711
runtime_overhead_inductor_requires_grad_dynamic,instruction_count,309979
runtime_overhead_inductor_requires_grad_backward_dynamic,instruction_count,77599
```

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames @Lucaskabela

### CODE REVIEW

File: benchmarks/dynamo/pr_time_benchmarks/benchmarks/runtime_overhead.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+    def name(self):
+        prefix = f"{self.category()}_{self.backend()}"
+        if self._requires_grad:
+            prefix += "_requires_grad"
+        if self._inference_mode:
+            prefix += "_inference_mode"
+        if self.is_dynamic():
+            prefix += "_dynamic"
+        return prefix
```
Comment (laithsakka) [inline benchmarks/dynamo/pr_time_benchmarks/benchmarks/runtime_overhead.py:36]: do we need a benchmark name before the prefix?

File: benchmarks/dynamo/pr_time_benchmarks/benchmarks/runtime_overhead.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+        # warmup
+        self._work()
+
+    def _prepare(self):
+        pass
+
+    def _work(self):
+        if self._inference_mode:
+            with inference_mode():
```
Comment (laithsakka) [inline benchmarks/dynamo/pr_time_benchmarks/benchmarks/runtime_overhead.py:69]: shall we run that in a loop? or is it on purpose that you want one invocation?

Comment (azahed98) [reply benchmarks/dynamo/pr_time_benchmarks/benchmarks/runtime_overhead.py:69]: I wasn't on certain on this, but I copied some of the the other benchmarks, where `_work` seems to be 1 "unit" of work. I also noticed `BenchmarkBase` has `_num_iterations`, so I assumed that would handle repeating.

File: benchmarks/dynamo/pr_time_benchmarks/benchmarks/runtime_overhead.py (hunk 1 above)
Comment (azahed98) [reply benchmarks/dynamo/pr_time_benchmarks/benchmarks/runtime_overhead.py:36]: I think this should be covered by the `prefix = f"{self.category()}_{self.backend()}"` line.

File: benchmarks/dynamo/pr_time_benchmarks/benchmarks/runtime_overhead.py (hunk 2 above)
Comment (jansel) [reply benchmarks/dynamo/pr_time_benchmarks/benchmarks/runtime_overhead.py:69]: Is the instruction count determinstic?

**Review by jansel — APPROVED** [review]

File: benchmarks/dynamo/pr_time_benchmarks/benchmarks/runtime_overhead.py (hunk 2 above)
Comment (azahed98) [reply benchmarks/dynamo/pr_time_benchmarks/benchmarks/runtime_overhead.py:69]: It is not, e.g.
```
collecting instruction count for runtime_overhead_inductor
instruction count for iteration 0 is 268847
instruction count for iteration 1 is 224156
instruction count for iteration 2 is 222669
instruction count for iteration 3 is 227857
instruction count for iteration 4 is 222645
```

**Review by mlazos — APPROVED** [review]

**Review by anijain2305 — APPROVED** [review]
Would suggest to warmup a little more in `_prepare_once`, just for safety (no real concern).
